packages/itwin/tree-widget/src/components/trees/models-tree/UseModelsTree.tsx

```diff
@@ -185,6 +208,36 @@ export function useModelsTree({ activeView, filter, hierarchyConfig, visibilityH
   };
 }
 
+async function getModels(paths: HierarchyFilteringPaths, idsCache: ModelsTreeIdsCache, classInspector: ECClassHierarchyInspector) {
+  if (!paths) {
+    return undefined;
+  }
+
+  const targetModels = new Set<Id64String>();
+  const targetSubjects = new Set<Id64String>();
+  for (const path of paths) {
+    const currPath = Array.isArray(path) ? path : path.path;
+    for (let i = 0; i < currPath.length; i++) {
+      const currStep = currPath[i];
+      if (!HierarchyNodeIdentifier.isInstanceNodeIdentifier(currStep)) {
+        break;
+      }
+
+      if (await classInspector.classDerivesFrom(currStep.className, "BisCore.GeometricModel3d")) {
+        targetModels.add(currStep.id);
+        break;
+      }
+
+      if (i === currPath.length - 1) {
+        targetSubjects.add(currStep.id);
+      }
```

Comment:
am I missing something, or is this adding everything except models to the `targetSubjects` set?